x-pack/plugin/core/src/main/java/org/elasticsearch/index/engine/RewriteCachingDirectoryReader.java

```diff
@@ -0,0 +1,267 @@
+                                throw new UnsupportedOperationException();
+                            }
+
+                            @Override
+                            public byte[] getMinPackedValue() {
+                                return minPackedValue;
+                            }
+
+                            @Override
+                            public byte[] getMaxPackedValue() {
+                                return maxPackedValue;
+                            }
+
+                            @Override
+                            public int getNumDataDimensions() {
+                                return numDataDimensions;
+                            }
+
+                            @Override
+                            public int getNumIndexDimensions() {
+                                return numDimensions;
+                            }
+
+                            @Override
+                            public int getBytesPerDimension() {
+                                return bytesPerDimension;
+                            }
+
+                            @Override
+                            public long size() {
+                                return size;
+                            }
+
+                            @Override
+                            public int getDocCount() {
+                                throw new UnsupportedOperationException();
+                            }
+                        });
+                    }
+                }
+            }
+            pointValuesMap = valuesMap;
+        }
+
+        @Override
+        public CacheHelper getCoreCacheHelper() {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public Terms terms(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public NumericDocValues getNumericDocValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public BinaryDocValues getBinaryDocValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public SortedDocValues getSortedDocValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public SortedNumericDocValues getSortedNumericDocValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public SortedSetDocValues getSortedSetDocValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public NumericDocValues getNormValues(String field) {
+            throw new UnsupportedOperationException();
+        }
+
+        @Override
+        public FieldInfos getFieldInfos() {
+            return new FieldInfos(new FieldInfo[0]);
```

Comment:
I think you need to handle the point values field here to make `FieldSubsetReader` happy ?